commons/com.b2international.index.api.tests/src/com/b2international/index/GroovyMemoryLeakTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.index;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Maps.newHashMap;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.Test;
+
+import com.b2international.index.Fixtures.Data;
+import com.b2international.index.query.Expressions;
+import com.b2international.index.query.Query;
+import com.b2international.index.query.SortBy;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * @since 5.10.11
+ */
+public class GroovyMemoryLeakTest extends BaseIndexTest {
+
+	private static final int NUM_DOCS = 10_000;
+	
+	@Test
+	public void tryToGenerateMemoryLeak() throws Exception {
```

Comment:
It might be worth noting here in a comment that the test may just "fizzle out" because of memory exhaustion instead of actually failing. Or does it now fail reliably with the time limit set?